app/self/page.tsx

```
@@ -2,82 +2,78 @@
+  }
 
-
-  // Update userId when wallet address changes
-  useEffect(() => {
-    if (address && isConnected) {
-      setUserId(address);
-    }
-  }, [address, isConnected]);
-
-  // Use useEffect to ensure code only executes on the client side
   useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)